src/app/components/wrapper-components/rules-configuration/rules-configuration.component.ts

```diff
@@ -0,0 +1,221 @@
+                click: MGPOptional.empty(),
+                name: this.gameName,
+                node,
+            };
+            this.cdr.detectChanges();
+        }
+    }
+
+    public getConfigDemo(): DemoNodeInfo {
+        return this.configDemo;
+    }
+
+    private generateForm(config: RulesConfig, configurable: boolean): void {
+        const group: ConfigFormJson = {};
+
+        Object.keys(config).forEach((parameterName: string) => {
+            const value: ConfigDescriptionType =
+                this.getRulesConfigDescriptionValue(parameterName,
+                                                    config[parameterName]);
+
+            group[parameterName] = this.getFormControl(parameterName, value, configurable);
+        });
+        this.rulesConfigForm = new FormGroup(group);
+    }
+
+    private assertParamsAreCoherent(): void {
+        if (this.userIsCreator === false) {
+            Utils.assert(this.rulesConfigToDisplay != null,
+                         'Config should be provided to non-creator in RulesConfigurationComponent');
+        }
+    }
+
+    private getRulesConfigDescriptionValue(name: string, defaultValue: ConfigDescriptionType): ConfigDescriptionType {
+        if (this.userIsCreator === false) {
+            const configuration: RulesConfig = Utils.getNonNullable(this.rulesConfigToDisplay);
+            defaultValue = configuration[name];
+        }
+        return defaultValue;
+    }
+
+    private getFormControl(parameterName: string, value: ConfigDescriptionType, configurable: boolean): FormControl {
+        let formControl: FormControl;
+        if (typeof value === 'number') {
+            const mgpValidators: MGPValidator = this.rulesConfigDescription.validator[parameterName];
+            const validator: CustomValidator = this.getNumberValidator(mgpValidators);
+            formControl = new FormControl(value, validator) as FormControl<number>;
+            formControl = new FormControl(value) as FormControl<number>;
+        } else {
+            Utils.expectToBe(typeof value, 'boolean');
+            formControl = new FormControl(value) as FormControl<boolean>;
+        }
+        if (configurable === false) {
+            formControl.disable();
+        }
+        formControl.valueChanges.subscribe(() => {
+            this.onUpdate();
+        });
+        return formControl;
+    }
+
+    private getNumberValidator(mgpValidator: MGPValidator): CustomValidator {
+        return (control: AbstractControl) => {
+            return { invalidTruc: mgpValidator(control.value).isFailure() };
+        };
+    }
+
+    public onUpdate(): void {
+        if (this.userIsCreator === false) {
+            ErrorLoggerService.logError('RulesConfiguration', 'Only creator should be able to modify rules config');
+        } else if (this.chosenConfig !== 'Custom') {
+            ErrorLoggerService.logError('RulesConfiguration', 'Only Customifiable config should be modified!');
```

Comment:
remplacer par des asserts ?

Reply:
erh, y a des bugs chiants, si t'insiste je fais le changement haha :D

Reply:
`Assert(this.userIsCreator, 'RulesConfiguration', 'Only creator ...')` rajoute des bugs ?

Reply:
TODOé, c'est chiant pour pas grand chose et en plus karma jasmine veulent pas faire leur boulot ça devient très violencigêne :D

Reply:
je testerai donc !